src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CurrentPeriodServiceImpl.java

```diff
@@ -1,33 +1,165 @@
+        currentPeriod.setEtag(GenerateEtagUtil.generateEtag());
+        currentPeriod.setKind(Kind.CURRENT_PERIOD.getValue());
+        CurrentPeriodEntity currentPeriodEntity = currentPeriodTransformer.transform(currentPeriod);
+
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+        debugMap.put("company_accounts_id", companyAccountId);
+
+        String id = generateID(companyAccountId);
+        currentPeriodEntity.setId(id);
+        debugMap.put("id", id);
+
+        try {
+            currentPeriodRepository.insert(currentPeriodEntity);
+        } catch (DuplicateKeyException dke) {
+            LOGGER.errorContext(requestId, dke, debugMap);
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR, null);
+        } catch (MongoException me) {
+            DataException dataException = new DataException(
+                    "Failed to insert " + ResourceName.SMALL_FULL.getName(), me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+
+        smallFullService.addLink(companyAccountId, LinkType.SMALL_FULL, selfLink, requestId);
+
+        return new ResponseObject<>(ResponseStatus.CREATED, currentPeriod);
     }
 
     @Override
-    public String getResourceName() {
-        return "current-period";
+    public ResponseObject<CurrentPeriod> findById(String id, String requestId)
+            throws DataException {
+        CurrentPeriodEntity currentPeriodEntity;
+        try {
+            currentPeriodEntity = currentPeriodRepository.findById(id).orElse(null);
+        } catch (MongoException me) {
+            final Map<String, Object> debugMap = new HashMap<>();
+            debugMap.put("id", id);
+            DataException dataException = new DataException("Failed to find Current period", me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+
+        if (currentPeriodEntity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+        CurrentPeriod currentPeriod = currentPeriodTransformer.transform(currentPeriodEntity);
+        return new ResponseObject<>(ResponseStatus.FOUND, currentPeriod);
+    }
+
+    @Override
+    public void addLink(String id, LinkType linkType, String link, String requestId)
+            throws DataException {
+        CurrentPeriodEntity currentPeriodEntity = currentPeriodRepository.findById(id)
+                .orElseThrow(() -> new DataException(
+                        "Failed to add get Current period entity to add link"));
+        CurrentPeriodDataEntity currentPeriodDataEntity = currentPeriodEntity.getData();
+        Map<String, String> map = currentPeriodDataEntity.getLinks();
+        map.put(linkType.getLink(), link);
+        currentPeriodDataEntity.setLinks(map);
+        currentPeriodEntity.setData(currentPeriodDataEntity);
```

Comment:
do we need this `setData ` call? we've just got the `currentPeriodDataEntity` from this object and added one to it so it's already set on the `currentPeriodEntity ` object